scripts/subscribeEventProof.js

```diff
@@ -117,26 +99,70 @@ async function main (networkName, bridgeContractAddress) {
     logger.info(`CENNZnet bridge deployed to: ${bridge.address}`);
     logger.info(`Executor: ${txExecutor.address}`);
 
-    let lastValidatorsSet = getValidatorAddedToBridge(bridge);
+    // For any kind of restart, we check if the last event proof generated on CENNZnet side, has been update on Eth side
+    logger.info('Check the last event proof generated on CENNZnet side, has been update on Eth side');
+    const lastEventProofIdFromCennznet = await api.query.ethBridge.notarySetProofId();
+    logger.info(`lastEventProofIdFromCennznet: ${lastEventProofIdFromCennznet.toString()}`);
+    const eventExistsOnEth = await bridge.eventIds(lastEventProofIdFromCennznet.toString());
+    logger.info(`eventExists on Ethereum: ${eventExistsOnEth}`);
+    try {
+        // check if event proof exist on Eth for last event proof id of CENNZnet
+        if (!eventExistsOnEth) {
+            const lastEventProcessed = await EventProcessed.findOne();
```

Comment:
is this guaranteed to return the 'last event processed'?

Reply:
yes.. It will only store the last eventId always.